tgui/packages/tgui/interfaces/MatMarket.js

```diff
@@ -0,0 +1,113 @@
+import { useBackend } from '../backend';
+import { Section, Stack, Flex, Button } from '../components';
+import { Window } from '../layouts';
+
+export const MatMarket = (props, context) => {
+  const { act, data } = useBackend(context);
+
+  const { materials = [] } = data;
+  return (
+    <Window width={675} height={400}>
+      <Window.Content scrollable>
+        <Section
+          title="Materials for sale"
+          buttons={
+            <Button
+              icon="dollar"
+              tooltip="Place order from cargo budget."
+              color={data.orderingPrive && data.canOrderCargo ? 'green' : null}
+              content="Order via Cargo Budget"
+              onClick={() => act('toggle_budget')}
+            />
+          }>
+          Buy orders for material sheets placed here will be ordered on the next
+          cargo shipment.
+          <br />
+          To sell materials, please insert sheets or similar stacks of
+          materials. All minerals sold on the market directly are subject to an
+          20% market fee.
+          <Section>
+            Current credit balance:{' '}
+            <b>{data.creditBalance ? data.creditBalance : 'zero'}</b> cr.
+          </Section>
+        </Section>
+        {materials.map((material) => (
+          <Section title={material.name} key={material.id}>
+            <Flex grow={1} basis={0}>
```

Comment:
not sure if this works here, but also i'd like to see only stack used unless you had an explicit reason for flex (yes you can nest stacks if that was the reason). the equivalent would be `Stack fill`